Parser/Types/Types.cs

```diff
@@ -6,7 +6,7 @@ namespace PSI;
 using static NType;
 
 // The data-type at any NExpr node
-public enum NType { Unknown, Int, Real, Bool, String, Char, Error, Void }
+public enum NType { None, Int, Real, Bool, String, Char, Error, Void }

```

Comment:
I think it's better to keep the *Unknown* enumeration as indicating we don't yet know the type. Later, when we determine it to be *None*, we can set that. For procedures, you could use the *Void* type I added at the end.